core/src/main/java/io/crate/types/CollectionType.java

```diff
@@ -72,10 +72,16 @@ public void writeTo(StreamOutput out) throws IOException {
         DataTypes.toStream(innerType, out);
     }
 
+    @Override
     public String getName() {
-        return innerType.getName();
+        String innerName = "";
+        if (innerType != null) {
```

Comment:
Can we enforce the innerType to be nonnull? `innerType()` isn't annotated as nullable, `isNumeric`, `isDecimal`, `compareTo` etc.  lack null checks.